.github/workflows/generate-query-help-docs.yml

```
@@ -15,10 +22,17 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - name: Clone github/codeql
+    - if: github.event.inputs.ref
+      name: Clone github/codeql (user defined ref)
       uses: actions/checkout@v2
       with:
-        path: codeql 
+        path: codeql
+        ref: ${{ github.event.inputs.ref }}
+    - if: github.event.inputs.ref == ''
+      name: Clone github/codeql (default ref)
+      uses: actions/checkout@v2
+      with:
+        path: codeql
```

Also not needed: if no `ref` is supplied to `actions/checkout`, it defaults to `github.ref` which will come from the dispatch event. The existing code is fine.
```suggestion
        path: codeql
```